### ISSUE TITLE
Setup LogisticRegression layer for multiple outputs

### ISSUE THREAD

**Issue by Mike43110** [dmlc/MXNet.jl#93]
I am not used to using this kind of library so I wanted to get a nice small network going.

Using the iris data set I want to classify data from `[Float, Float, Float, Float]` to `[Float, Float, Float]`.

So I set up the column vectors for the inputs and expected outputs and fed them into `ArrayDataProvider`. However, this only results in a single output from probs.

```julia
mlp = @mx.chain mx.Variable(:i) =>
  mx.MLP([4,7,3], hidden_activation=:sigmoid) =>
  mx.LogisticRegressionOutput(name=:o)

train_dataset = mx.ArrayDataProvider(:i => tt,
  :o_label => ot, shuffle=true, batch_size=3)

model = mx.FeedForward(mlp, context=mx.cpu())

  optimiser = mx.SGD(lr=0.1, momentum=0.9, weight_decay = 0)
  mx.fit(model, optimiser, train_dataset, eval_data=eval_dataset, n_epoch = 100)

  probs = mx.predict(model, eval_dataset)[1]
```

Is the code used. 
`size(tt) = (4,75)` `size(ot) = (3,75)` are the dimensions of the data and label arrays.

I am unsure how to get a multiple outputs from this.

Using `SoftMaxOutput(name=:o, multi_output=true)` in its place gives instead shape inconsistent, `Provided=(3,3)inferred shape = (3,1)`.

Unsure where to go from here, any help would be appreciated.

UPDATE:
I found grouping and added that. I split the output labels into 3 1D vectors for each output and added a fully connected layer behind each regression output:
```julia
  mlp = @mx.chain mx.Variable(:i) =>
  mx.MLP([4,7,3], hidden_activation=:sigmoid)

  o1 = @mx.chain mx.MLP(mlp,[(1,:sigmoid)]) =>
    mx.LinearRegressionOutput(name=:o1)
  o2 = @mx.chain mx.MLP(mlp,[(1,:sigmoid)]) =>
    mx.LinearRegressionOutput(name=:o2)
  o3 = @mx.chain mx.MLP(mlp,[(1,:sigmoid)]) =>
    mx.LinearRegressionOutput(name=:o3)
```

The input to o1,o2,o3 is now a 1D vector giving the expected output from each column. However, I still seem to have messed up somewhere. Training has the same accuracy for every epoch and the output is the same for each prediction value.

**Comment by vchuravy** [comment]
If I run our code probs has the shape (3, 75) and so to get the multidimensional output `probs[:, 1]` should do the trick.

**Comment by Mike43110** [comment]
I must have been extremely tired, no idea how I kept on seeing the displayed output as a 1D vector.
I got it working using grouping as an alternative. Which is a good thing as grouping is more what I needed in the 1st place. I need 2 true or false values and 4 linear outputs. So this was not a wasted endeavor for me.

Thanks for the help though. I apologise for wasting your time! That will teach me to try new libraries half asleep...
